src/Fixer/ClassNotation/FinalInternalClassFixer.php

```diff
@@ -202,12 +260,93 @@ private function isClassCandidate(TokensAnalyzer $tokensAnalyzer, Tokens $tokens
             $tags[$tag] = true;
         }
 
-        foreach ($this->configuration['annotation_include'] as $tag => $true) {
-            if (!isset($tags[$tag])) {
-                return false; // ignore class: class-level PHPDoc does not contain all tags that has been included through configuration
+        return $this->isConfiguredAsInclude($tags);
+    }
+
+    private function isClassCandidateBasedOnAttribute(Tokens $tokens, int $index): bool
+    {
+        $attributeCandidates = [];
+
+        while ($tokens[$index]->isGivenKind(CT::T_ATTRIBUTE_CLOSE)) {
```

Comment:
Can we add test case like this?

```php
#[A]
/**
 * @B
 */
#[C]
class Foo {}
```

I really hope no-one is coding like this, but looking at the current implementation I think it would only catch `C` attribute and wouldn't reach `A`, so it would be really great if we handle it properly 🙂. Or such test case may prove me wrong, which would be good too 😉.